elasticdl/python/tests/worker_ps_pytorch_test.py

```diff
@@ -0,0 +1,209 @@
+                DistributionStrategy.PARAMETER_SERVER,
+            ]
+            args = parse_worker_args(arguments)
+            worker = WorkerPytorch(args, ps_client=PSClient(self._channels))
+            self._workers.append(worker)
+
+    def _worker_train(
+            self, worker_id, train_db, test_db, stop_step
+    ):
+        # TODO: acc_meter with PyTorch
+        acc_meter = tf.keras.metrics.Accuracy()
+
+        worker = self._workers[worker_id]
+        worker_results = []
+        for step, (x, y) in enumerate(train_db):
+            if step == 0:
+                worker._run_model_call_before_training(x)
+            worker.get_model()
+            w_loss, w_grads = worker.training_process_pytorch(x, y)
+            worker.report_gradient(w_grads)
+
+            if step % 20 == 0:
+                worker.get_model()
+                for (x, y) in test_db:
+                    out = worker.forward_process(x)
+                    if "mnist" in self._model_def:
+                        acc_meter.update_state(torch.argmax(out, dim=1).numpy(), y.numpy())
+                    else:
+                        break
+                worker_results.append(
+                    (float(w_loss.detach().numpy()), float(acc_meter.result().numpy()))
+                )
+                acc_meter.reset_states()
+
+            if step > stop_step:
+                break
+        return worker_results
+
+    def _dataset_pytorch(self, dataset, batch_size):
+        """
+        _dataset_fn() builds dataset for TensorFlow
+        this func transforms dataset and set DataLoader for PyTorch
+        TODO: rewrite _dataset_fn() by IterableDataSet for PyTorch
+        """
+        dataset_list = []
+        for data_enum in list(dataset.as_numpy_iterator()):
+            shape_0 = data_enum[0].shape[0]
+            for i in range(shape_0):
+                dataset_list.append((data_enum[0][i:i + 1, ...], data_enum[1][i, ...]))
+        # Focus: labels.shape [batch_size] , [batch_size, 1] is incorrect
+
+        iterable_dataset = CustomDataset(dataset_list)
+        dataloader = DataLoader(dataset=iterable_dataset, batch_size=batch_size)
+        return dataloader
+
+    def test_compare_mnist_train(self):
+        model_def = "mnist.mnist_subclass_pytorch.CustomModel"
+        self._create_pserver(model_def, 2)
+
+        # Reuse the tf style code to get dataset
+        db, test_db = get_mnist_dataset(self._batch_size)
+        db = self._dataset_pytorch(db, self._batch_size)
+        test_db = self._dataset_pytorch(test_db, self._batch_size)
+
+        stop_step = 20
+        self._create_worker(1)
+        worker_results = self._worker_train(
+            0, train_db=db, test_db=test_db, stop_step=stop_step
+        )
+        print("show batchs train worker_results[0]\n"
+              "w_loss, acc_meter:", worker_results[0])
+        print("--------finish test_compare_mnist_train!")
+
+    def test_compare_onebatch_train_pytorch(self):
```

Comment:
We can remove this unit test since it does not do comparing.

Reply:
get it.